ui/tests/components/table/table-element/ColumnSettings.test.tsx

```diff
@@ -0,0 +1,118 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import ColumnSettings from '@/components/table/table-element/ColumnSettings';
+import { Table } from '@tanstack/table-core';
+import { GroupingPath } from '@/models/groupings-api-results';
+
+const mockVisibilityState = {
+    DESCRIPTION: true,
+    'GROUPING PATH': true,
+};
+
+const mockToggleVisibility = jest.fn((columnKey: 'DESCRIPTION' | 'GROUPING PATH', newState) => {
+    mockVisibilityState[columnKey] = newState;
+});
+
+const columnMocks = {
+    DESCRIPTION: {
+        toggleVisibility: (newState: boolean) => mockToggleVisibility('DESCRIPTION', newState),
+    },
+    'GROUPING PATH': {
+        toggleVisibility: (newState: boolean) => mockToggleVisibility('GROUPING PATH', newState),
+    },
+};
+
+const mockTable = {
+    getColumn: jest.fn().mockImplementation((columnKey: 'DESCRIPTION' | 'GROUPING PATH') => {
+        return {
+            toggleVisibility: columnMocks[columnKey].toggleVisibility,
+        };
+    }),
+} as unknown as Table<GroupingPath>;
```

Comment:
I think this can be simplified to:
```
const mockColumnVisibility = {
    DESCRIPTION: true,
    'GROUPING PATH': true
};

const mockTable = {
    getColumn: (columnKey: 'DESCRIPTION' | 'GROUPING PATH') => ({
        toggleVisibility: () => {
            mockColumnVisibility[columnKey] = !mockColumnVisibility[columnKey];
        }
    })
} as Table<GroupingPath>;
```
So instead using `expect(mockToggleVisibility).toHaveBeenCalledWith('DESCRIPTION', false);`, you could do
`expect(mockVisibilityState['DESCRIPTION']).toBeFalsy()`

Reply:
done